### ISSUE TITLE
Recognize missing brackets for assignment expression

### ISSUE THREAD

**Issue by cdce8p** [microsoft/pylance-release#1838]
It would be nice if `Pylance` could recognize that the example below would result in a `SyntaxError`.

```py
l = lambda s: val if val := s.capitalize() else None
```

```
$ python test.py
  File "/.../test.py", line 1
    l = lambda s: val if val := s.capitalize() else None
                  ^^^^^^^^^^
SyntaxError: expected 'else' after 'if' expression
```

--
To be valid, brackets around the assignment expression are necessary. This only applies to if expressions. Normal if statements will work, even without them.
```py
l = lambda s: val if (val := s.capitalize()) else None
```

--
Tested with Pylance: v2021.9.2

**Comment by erictraut** [comment]
Thanks for the bug report. There was indeed a bug in pyright's parser that resulted in a false negative in this case. The fix will be in the next release.

**Comment by jakebailey** [comment]
This issue has been fixed in version 2021.9.3, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202193-22-september-2021
